pogobuf/pogobuf.client.js

```diff
@@ -953,23 +979,40 @@ function Client() {
         // If the requests include a map objects request, make sure the minimum delay
         // since the last call has passed
         if (requests.some(r => r.type === RequestType.GET_MAP_OBJECTS)) {
+
+            // Check distance traveled to see if it is past refresh distance
+            var refreshRequired = !self.withinInteractionRange(
+                self.lastMapObjectsLatitude,
+                self.lastMapObjectsLongitude,
+                self.mapObjectsMinDistance);
             var now = new Date().getTime(),
-                delayNeeded = self.lastMapObjectsCall + self.mapObjectsMinDelay - now;
+                delayNeeded = self.lastMapObjectsCall + self.mapObjectsMaxDelay - now;
```

Comment:
This would better mimic the app, which I think is what you were going for here anyway.  When standing still, my logs show a map objects request every 30 seconds. It's when your moving that it becomes more frequent, which is handled by the refresh required logic.